aya/src/bpf.rs

```diff
@@ -58,7 +60,243 @@ pub(crate) struct bpf_map_def {
     pub(crate) map_flags: u32,
     // optional features
     pub(crate) id: u32,
-    pub(crate) pinning: u32,
+    pub(crate) pinning: BpfPinningType,
+}
+
+#[repr(u32)]
+#[derive(Copy, Clone, Debug, PartialEq)]
+pub(crate) enum BpfPinningType {
+    PinNone = 0,
+    PinByName = 1,
+}
+
+impl TryFrom<u32> for BpfPinningType {
```

Comment:
I don't think this is ever used? If so I would remove it and remove `UnexpectedPinningType` too